src/core/image/image.service.test.ts

```diff
@@ -0,0 +1,53 @@
+import { ImageService } from './image.service';
+import { getTestImage, getTestImageBuffer, streamToBuffer } from '../../../test/test.utils';
+import type { Region } from 'sharp';
+import sharp from 'sharp';
+
+describe('ImageService', () => {
+  let imageService: ImageService;
+  beforeEach(() => {
+    imageService = new ImageService();
+  });
+
+  describe('#resize', () => {
+    it('should resize passed image', async () => {
+      const expectedDimensions = 20;
+      const image = await getTestImage();
+
+      const resizedImage = imageService.resize(image, {height: expectedDimensions});
+
+      const resizedImageMetadata = await sharp(await streamToBuffer(resizedImage)).metadata();
+      expect(resizedImageMetadata).toMatchObject({
+        width: expectedDimensions,
+        height: expectedDimensions,
+      });
+    });
+  });
+
+  describe('#crop', () => {
+    it('should crop passed image', async () => {
+      const expectedDimensions = 1;
+      const cropOptions = {top: expectedDimensions, height: expectedDimensions, width: expectedDimensions, left: expectedDimensions} as Region;
+      const image = await getTestImage();
+
+      const croppedImage = imageService.crop(image, cropOptions);
+
+      const croppedImageMetadata = await sharp(await streamToBuffer(croppedImage)).metadata();
+      expect(croppedImageMetadata).toMatchObject({
```

Comment:
Actually, I don't know how to test this well, but you are testing `width, height` properties of cropping here, but not testing `top`/`left` working properly.

I think you can take pixel color of pixel in [0, 0] or something

Reply:
I was thinking about that, but it looks like testing external library. I can write another test to just check if parameters are passed properly.

Reply:
Yeah, but if external lib will break, our will as well, and we would to catch this for example if we update sharp or something. It doesn't matter that it's external, we must to have this tested

Reply:
Of course it's not about testing whole API, but only smoke test